File: java/src/Stopwatch.java
public class Stopwatch {
  private int timeLapsed;
  
  public Stopwatch() {
    timeLapsed = 0;
  }
  
  public void start() {
    while (true) {
        int[] time = convertFromSeconds(timeLapsed);
        printTime(time);
        wait(1000);
        clearLine();
        timeLapsed++;
    }
  }
  
  private void wait(int ms) {
    try {
        Thread.sleep(ms);
    } catch (InterruptedException e) {
        Thread.currentThread().interrupt();
    }
  }

  private int[] convertFromSeconds(int secs) {
      int mins = secs / 60;
      secs %= 60;
      int hrs = mins / 60;
      mins %= 60;
      int[] time = {hrs, mins, secs};
      return time;
  }

  private void printTime(int[] time) {
      StringBuilder sb = new StringBuilder();
      sb.append(String.format("%02d", time[0]));
      sb.append(":");
      sb.append(String.format("%02d", time[1]));
      sb.append(":");
      sb.append(String.format("%02d", time[2]));
      System.out.println(sb.toString());
  }

  private void clearLine() {
    System.out.print("\033[F");
  }
}

File: kotlin/src/Stopwatch.kt
class Stopwatch() {
    fun start() {
		var timeLapsed = 0
		while (true) {
			clearLine()
			val (hrs, mins, secs) = convertFromSeconds(timeLapsed)
			printTime(hrs, mins, secs)
			waitFor(1000)
			timeLapsed++
		}
	}

	private fun convertFromSeconds(secs: Int): Triple<Int, Int, Int> {
		var mins = secs / 60
		val remSecs = secs % 60
		val hrs = mins / 60
		mins %= 60
		return Triple(hrs, mins, remSecs);
	}

	private fun printTime(hrs: Int, mins: Int, secs: Int) {
		val twoDigit = fun (num: Int) = (if (num < 10) "0" else "") + num
		print("${twoDigit(hrs)}:${twoDigit(mins)}:${twoDigit(secs)}")
	}

	private fun clearLine() {
		print("\b\b\b\b\b\b\b\b")
	}

	private fun waitFor(ms: Long) {
		try {
			Thread.sleep(ms);
		} catch (e: InterruptedException) {
			Thread.currentThread().interrupt();
		}
	}
}

fun main() = Stopwatch().start()
